[FIX][12.0][account_move_transactionid_import] s/transaction_ref/ref

Tonow-c2c opened the pull request:
the field `transaction_ref` doesn't exist any more

Tonow-c2c committed:
[FIX][12.0][account_move_transactionid_import] s/transaction_ref/ref

Remove 'ref' from account.move.line preparation

As 'ref' is related to account.move.ref it is not needed

Fix multi move import

multi_move_import on the parser allows to generate 1 account.move
for each line in the imported file, instead of having all the lines
on the same account.move

grindtildeath reviewed (changes requested):
Doesn't work as expected, let's check it out together when you're back.

grindtildeath committed:
Use multi move import when importing using transaction_id

With the drop of 'transaction_ref' field on account.move.line in
order to ease the reconcilition using 'ref' on account.move, we
need to import each line from the imported file in different
account.moves.

grindtildeath commented:
@Tonow-c2c https://github.com/Tonow-c2c/account-reconcile/pull/1 :wink:

i-vyshnevska commented on account_move_base_import/parser/file_parser.py:
```diff
@@ -70,13 +72,24 @@ def _parse(self, *args, **kwargs):
+            if len(self.parsed_file) <= self.current_line:
+                return False
+            else:
+                print(self.current_line)
```
```suggestion
```

i-vyshnevska commented on account_move_transactionid_import/parser/transactionid_file_parser.py:
```diff
@@ -32,6 +32,7 @@ def __init__(self, profile, ftype='csv', extra_fields=None, header=None,
         super().__init__(
             profile, extra_fields=conversion_dict, ftype=ftype, header=header,
             **kwargs)
+        self.support_multi_moves = True
```
should be optional?

i-vyshnevska replied:
I don't really understand the message of one commit:
 "parser allows to generate 1 account.move for each line in the imported file"
account move should have at least 2 lines, and second is searched/created on statement reconciliation 
Lint need to be checked

grindtildeath replied:
> should be optional?

Not when using transaction_id parser due to the change from #292 

> "(support_multi_moves) parser allows to generate 1 account.move for each line in the imported file"
account move should have at least 2 lines, and second is searched/created on statement reconciliation

The `support_multi_moves` flag on the parser allows to generate 1 account.move per line in the imported statement. However resulting account move (for one line) will always include at least two move lines (cf https://github.com/OCA/account-reconcile/blob/12.0/account_move_base_import/models/account_journal.py#L319-L324) and even 3 if there's a commission amount (cf https://github.com/OCA/account-reconcile/blob/12.0/account_move_base_import/models/account_journal.py#L325-L326).

@Tonow-c2c Could you fix the lint and commit @i-vyshnevska 's suggestion ?

grindtildeath commented on account_move_base_import/models/account_journal.py:
```diff
@@ -245,7 +245,6 @@ def prepare_move_line_vals(self, parser_vals, move):
             'debit_cash_basic': values['debit'],
             'credit_cash_basic': values['credit'],
             'balance_cash_basic': values['debit'] - values['credit'],
```
@Tonow-c2c Could you please also remove these undefined fields ?

Tonow-c2c committed:
Remove undefined fields balance_cash credit_cash debit_cash

Tonow-c2c committed:
Pass Black on the files + pylint

Move test data to tests/data

simahawk commented:
@yvaucher quick check and merge?

yvaucher commented:
/ocabot merge major
